htdocs/api/v0.0.3-dev/MedhubConsent.php

```diff
@@ -0,0 +1,192 @@
+<?php
+
+
+namespace Loris\API;
+use http\Env\Response;
+
+set_include_path(get_include_path() . ":" . __DIR__);
+require_once 'APIBase.php';
+
+
+
+class MedhubConsent extends APIBase {
+
+
+
+    var $requestData;
+
+
+    /**
+     *  Candidates request handler
+     *
+     * @param string $method The HTTP request method of the request
+     * @param array  $data   The data that was POSTed to the request
+     */
+
+
+
+    public function __construct($method, $data=null)
+    {
+
+
+
+
+        $this->AllowedMethods = [
+            'PUT'
+        ];
+
+        $this->RequestData    = $data;
+
+        parent::__construct($method);
+    }
+
+
+
+    public function handlePUT()
+    {
+
+        $token = null;
+        $consentList = null;
+        $candid = null;
+
+        $data   = $this->RequestData;
+        if ($data === null) {
+            $this->header("HTTP/1.1 400 Bad Request");
+            $this->error("Can't parse data");
+            $this->safeExit(0);
+        }
+
+
+        if (!isset($data['Token'])) {
+            $this->header("HTTP/1.1 400 Bad Request");
+            $this->error("There is no Token object in the PUT data");
+            $this->safeExit(0);
+        }
+
+        if (!isset($data['ConsentList'])) {
+            $this->header("HTTP/1.1 400 Bad Request");
+            $this->error("There is no Consent List object in the PUT data");
+            $this->safeExit(0);
+        }
+
+        //Tries to select CandID + entry date for the given token --> later throws and error and dies if nothing found
+        $token = $data['Token'];
+        $consentList = $data['ConsentList'];
+
+        $candidTimeUsed = $this->DB->pselectRow(
+            "SELECT CandidateID, EntryDate, AlreadyUsed
+                FROM medhub_token
+             WHERE Token = :to
+                ",
+            ['to' => $token]
+        );
+
+
+        //Check that every consent is represented AND all fields are set
+        $consentNames = ['INF','HR', 'GEN', 'CL', 'CQ', 'CS', 'CR', 'ATPSY','ORPS'];
+        foreach($consentNames as $conName){
+            if (!isset($consentList[$conName]['Response'], $consentList[$conName]['Date'])){
+                error_log('Incomplete Consent Data');
+                $this->header("HTTP/1.1 400 Bad Request");
+                $this->safeExit(0);
+            }
+        }
+
+
+        if (!isset($candidTimeUsed['EntryDate'],$candidTimeUsed['CandidateID'] )or isset($candidTimeUsed['AlreadyUsed'])){
```

Comment:
`or` ??? in PHP its `||` that should not pass testing

Reply:
Also, dont check for isset(AlreadyUsed) because it will most likely be set to empty string or false or something, its too unpredictable. check directly for the value you are expecting i.e. `$candidTimeUsed['AlreadyUsed'] === 'TRUE'`